pkg/operator/status.go

```diff
@@ -323,3 +322,105 @@ func conditionTrue(conditions []configv1.ClusterOperatorStatusCondition, condTyp
 
 	return false
 }
+
+// updateConditions updates the specific condition to have status True and all others conditions to have status False
+// the only exception is Available condition because we should update the Upgradeable condition to True as well
+func updateConditions(
+	conditions []configv1.ClusterOperatorStatusCondition,
+	conditionType configv1.ClusterStatusConditionType,
+	reason string,
+	message string,
+) {
+	now := metav1.Now()
+	for i := range conditions {
+		c := &conditions[i]
+		if c.Type == conditionType {
+			conditions[i] = configv1.ClusterOperatorStatusCondition{
+				Type:               c.Type,
+				Status:             configv1.ConditionTrue,
+				LastTransitionTime: now,
+				Reason:             reason,
+				Message:            message,
+			}
+			continue
+		}
+
+		// Once we update the available condition we also should update the cluster operator to be upgradeable except the unmanaged state.
```

Comment:
Do you mean:
`s/be upgradeable except the unmanaged state/be upgradeable unless the operator is in Unmanaged state/`
?  In any case, I still think we probably do not want to set the `OperatorUpgradeable` condition.